File: adidas-ecommerce/services/cart-service/internal/models/gravatar.go
package models

import (
	"crypto/md5"
	"fmt"
	"strings"
)

func GenerateGravatarURL(email string) string {
	email = strings.ToLower(strings.TrimSpace(email))
	hash := md5.Sum([]byte(email))
	return fmt.Sprintf("https://www.gravatar.com/avatar/%x?d=identicon&s=80", hash)
}

File: apps/web/utils/gravatar.ts
import md5 from "md5"

export const getGravatarUrl = (email?: string, fallbackUrl = "/avatar-placeholder.png") => {
  if (!email) return fallbackUrl
  const hash = md5(email.trim().toLowerCase())
  return `https://www.gravatar.com/avatar/${hash}?d=identicon`
}
